### PR TITLE
allow test image to be customized

### PR DESCRIPTION
**Pull request by mvinkler**
## Description
Allow the kn test image to be customized via environment variable.

## Changes

* all references to KnDefaultTestImage were replaced with GetKnTestImage() function
* all hardcoded occuerences of "gcr.io/knative-samples/helloworld-go" were replaced with GetKnTestImage() function

### CODE REVIEW

File: lib/test/integration.go [hunk 1]
```diff
@@ -42,6 +43,14 @@ type KnTest struct {
 	kn        Kn
 }
 
+func GetKnTestImage() string {
```
[the remark this answers is not in the document]
Comment (mvinkler) [reply lib/test/integration.go:47]: Added comment.

File: test/e2e/service_file_test.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func TestServiceCreateFromFile(t *testing.T) {
 	defer os.RemoveAll(tempDir)
 	assert.NilError(t, err)
 
-	test.CreateFile("foo.json", fmt.Sprintf(ServiceJSON, test.KnDefaultTestImage), tempDir, test.FileModeReadWrite)
+	test.CreateFile("foo.json", fmt.Sprintf(ServiceJSON, test.GetKnTestImage()), tempDir, test.FileModeReadWrite)
 	test.CreateFile("foo.yaml", fmt.Sprintf(ServiceYAML, test.KnDefaultTestImage), tempDir, test.FileModeReadWrite)
```
Comment (daisy-ycguo) [inline test/e2e/service_file_test.go:91]: Do you need to change `test.KnDefaultTestImage` to `GetKnTestImage()` in this line?

Comment (mvinkler) [reply test/e2e/service_file_test.go:91]: I changed all occurrences,  is this line special?

Comment (navidshaikh) [reply test/e2e/service_file_test.go:91]: looks like thats the only left over occurrence @mvinkler

Comment (mvinkler) [reply test/e2e/service_file_test.go]: Ah, I see. Totally missed that one. Thank you.

**Pull request merged**